diff --git a/node_modules/gatsby-source-notionso/lib/renderer/renderNotionBlocks.js b/node_modules/gatsby-source-notionso/lib/renderer/renderNotionBlocks.js
index 58f7a49..496c0cf 100644
--- a/node_modules/gatsby-source-notionso/lib/renderer/renderNotionBlocks.js
+++ b/node_modules/gatsby-source-notionso/lib/renderer/renderNotionBlocks.js
@@ -124,8 +124,10 @@ function renderBlock(root, renderFuncs, renderUtils, debug) {
         if (!url) {
             console.log(`cannot find public url for image: ${JSON.stringify(meta)}`);
         }
+        const fluid = renderUtils.childImage(meta.source);
         // TODO: default image?
         meta.publicImageUrl = url || '';
+        meta.childImage = fluid || '';
     }
     if (hasProperty(root, 'title') && root.type !== 'page') {
         const texts = findTextProperty(root, 'title');
diff --git a/node_modules/gatsby-source-notionso/lib/renderer/renderUtils.js b/node_modules/gatsby-source-notionso/lib/renderer/renderUtils.js
index 29bc7b5..54fae6f 100644
--- a/node_modules/gatsby-source-notionso/lib/renderer/renderUtils.js
+++ b/node_modules/gatsby-source-notionso/lib/renderer/renderUtils.js
@@ -9,6 +9,13 @@ function default_1(imageNodes) {
             }
             return node.localFile && node.localFile.publicURL;
         },
+        childImage(imageUrl) {
+            const node = imageNodes.find(i => i.imageUrl === imageUrl);
+            if (!node) {
+                return null;
+            }
+            return node.localFile && node.localFile.childImageSharp.fluid;
+        }
     };
 }
 exports.default = default_1;
